pkg/cmd/admin/registry/registry.go

```diff
@@ -230,6 +233,11 @@ func RunCmdRegistry(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg
 			"OPENSHIFT_INSECURE":  insecure,
 		}
 
+		if len(ports) > 0 {
+			env["REGISTRY_HTTP_ADDR"] = fmt.Sprintf(":%d", ports[0].ContainerPort)
+			env["REGISTRY_HTTP_NET"] = "tcp"
```

Comment:
I know, but I thought it was better to enforce it because otherwise the behavior will not be such as we have documented. I mean we will have a bug if http.type will be changed in config.yml .